src/connectors/impls/clickhouse.rs

```diff
@@ -0,0 +1,290 @@
+    fn connection_url(&self) -> String {
+        let url = self.config.url.as_str();
+        let compression = self.config.compression.unwrap_or_default();
+
+        format!("{url}?compression={compression}")
+    }
+}
+
+#[derive(Deserialize)]
+struct ClickhouseConfig {
+    url: Url<ClickHouseDefaults>,
+    compression: Option<Compression>,
+    columns: Vec<Column>,
+}
+
+impl ConfigImpl for ClickhouseConfig {}
+
+#[derive(Clone, Copy, Deserialize)]
+#[serde(rename_all = "snake_case")]
+enum Compression {
+    None,
+    Lz4,
+}
+
+impl Default for Compression {
+    fn default() -> Compression {
+        Compression::None
+    }
+}
+
+impl Display for Compression {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            Compression::None => "none",
+            Compression::Lz4 => "lz4",
+        }
+        .fmt(f)
+    }
+}
+
+#[derive(Deserialize)]
+struct Column {
+    name: String,
+    #[serde(rename = "type")]
+    type_: DummySqlType,
+}
+
+pub(crate) struct ClickhouseSink {
+    db_url: String,
+    pool: Option<Pool>,
+    columns: Vec<(String, DummySqlType)>,
+}
+
+#[async_trait::async_trait]
+impl Sink for ClickhouseSink {
+    async fn connect(&mut self, _ctx: &SinkContext, _attempt: &Attempt) -> Result<bool> {
+        let pool = Pool::new(self.db_url.as_str());
+        self.pool = Some(pool);
+
+        Ok(true)
+    }
+
+    async fn on_event(
+        &mut self,
+        _input: &str,
+        event: Event,
+        _ctx: &SinkContext,
+        _serializer: &mut EventSerializer,
+        _start: u64,
+    ) -> Result<SinkReply> {
+        let mut client = self
+            .pool
+            .as_ref()
+            .ok_or_else(|| Error::from(ErrorKind::NoClickHouseClientAvailable))?
+            .get_handle()
+            .await?;
+
+        let (event_min_size, event_max_size) = event.value_iter().size_hint();
+        let block_size_estimate = event_max_size.unwrap_or(event_min_size);
+
+        let mut block = Block::with_capacity(block_size_estimate);
+
+        for value in event.value_iter() {
+            let row = self.clickhouse_row_of(value)?;
+            block.push(row)?;
+        }
+
+        debug!("Inserting block:{:#?}", block);
+        client.insert("people", block).await?;
+
+        Ok(SinkReply::NONE)
+    }
+
+    fn auto_ack(&self) -> bool {
+        true
+    }
+}
+
+impl ClickhouseSink {
+    fn clickhouse_row_of(
+        &self,
+        input: &tremor_value::Value,
+    ) -> Result<Vec<(String, clickhouse_rs::types::Value)>> {
+        let mut rslt = Vec::new();
+
+        let object = input
+            .as_object()
+            .ok_or_else(|| Error::from(ErrorKind::ExpectedObjectEvent(input.value_type())))?;
+
+        for (column_name, expected_type) in self.columns.iter() {
+            let cell = object.get(column_name.as_str()).ok_or_else(|| {
+                Error::from(ErrorKind::MissingEventColumn(column_name.to_string()))
+            })?;
```

Comment:
We could just use null instead of returning an error

Reply:
That would only work if the type is nullable, right? If not we would get a conversion error?

Reply:
Correct; this is handled by the Tremor value -> clickhouse value function

Reply:
After sketching a few solutions, I see no simple way of doing this. Given how small the UX impact is, I think the best thing to do is to just leave it *as is*.